### DISCUSSION TITLE
Prisma Raw query,  multiple string interpolation in Prisma Raw query

### DISCUSSION THREAD

**Discussion by Akash1298** [prisma/prisma#14405] [Q&A]
I have implemented multiple interpolation in query all are giving me a fine result, but only when there is distance in query it is giving a error in response.

I have pasted the error image also that i am getting in response.
<img width="1293" alt="Screenshot 2022-07-20 at 7 48 46 PM" src="https://user-images.githubusercontent.com/65884746/180005497-00e8c70f-4db7-4843-9e18-32fc0d0322aa.png">
<img width="960" alt="Screenshot 2022-07-20 at 7 48 31 PM" src="https://user-images.githubusercontent.com/65884746/180005448-2f39cbc3-5169-4641-a687-cb92bc440f78.png">

**Comment by austincrim** [comment]
Hey @Akash1298 👋,

[This Stack Overflow thread](https://stackoverflow.com/questions/28308418/postgres-org-postgresql-util-psqlexception-error-insufficient-data-left-in-me) suggests it might be due to a null character present in the query.

Are you certain all the parameters you are passing have some data in them?

**Reply by Akash1298** [reply]
Yes, all parameters that i am passing, i have checked them by console.log they are same and fine in both condition when there is distance available or not.

Basically, In Prisma sql this is only difference
```
This is valid:-  Prisma.sql`50000`
This is not valid:-  Prisma.sql`${'50000'}`

//If iam using valid one i am getting result in response but if i use the 2nd one i.e. invalid gives me error
```

```
console.log(Prisma.sql`${'50000'}`)
OUTPUT:- { text: '$1', sql: '?', values: [ '50000' ] }

console.log(Prisma.sql`50000`)
OUTPUT:-  { text: '50000', sql: '50000', values: [] } 
```

**Reply by austincrim** [reply]
I'm not sure you need to pass `distance` through `Prisma.sql`. Have you tried passing it directly to the final `queryRaw`?

**Reply by Akash1298** [reply]
Yes, I have tried to pass distance directly to `queryRaw` , but that doesn't works.

**Reply by austincrim** [reply]
Do you get an error? What doesn't work about it?

**Reply by Akash1298** [reply]
Yes, in response i am getting error, above i have added the screenshot.

**Reply by austincrim** [reply]
Could you share your full Prisma schema as well? I'm going to try to fully reproduce this.

**Reply by Akash1298** [reply]
Schema is similar to this, and also need to add `earthdistance` extension `CREATE EXTENSION earthdistance CASCADE;`

```
model Product {
  id      Int      @id @unique @default(autoincrement())
  title String?
  is_sold Boolean
  is_deleted Boolean @default(false)
  latitude Float
  longitude Float 
  category_id  Int
  user_id  Int
  createdAt DateTime @default(now())
}
```

**Reply by austincrim** [reply]
A couple of notes from trying to reproduce this:

1. on line 26 of the code in the screenshot, I had to remove the extra `${queryForDistance}`
2. I had no problems directly passing `queryForDistance` to `$queryRaw` without using `Prisma.sql`, but it has to be passed as an integer, not a string.

Have you tried passing `queryForDistance` directly to `$queryRaw` as an integer?

```ts
let queryForDistance = 50000
```

**Reply by Akash1298** [reply]
Yes, I have tried it as integer the same error (i.e. mentioned above) i got.

On line 26 I think you're talking about `${queryForCategory}` .

**Reply by austincrim** [reply]
I think the best way forward for now is to file a [bug report](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) with very specific and thorough instructions on how we can reproduce this, ideally with a sample repository.

**Reply by Akash1298** [reply]
Okay will create a sample repo, this weekend.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
